### PR TITLE
emit disconnected state update when device gets disconnected

### LINKED ISSUE #118: no connectedDeviceStream updates after connectToDevice subscription cancel
**Describe the bug**
If i listen to the ```connectedDeviceStream``` of a ```FlutterReactiveBle```instance, connect to a peripheral with ```connectToDevice``` and cancel the subscription to disconnect the peripheral later, the ```connectedDeviceStream``` notifies no ```ConnectionStateUpdate``` with the states ```DeviceConnectionState.disconnecting``` & ```DeviceConnectionState.disconnected```.

**To Reproduce**
```Dart
...

final _ble = FlutterReactiveBle();
_stateUpdateSubscription = _ble.connectedDeviceStream.listen(_onConnectDevice);
_connectionStateUpdateSubscription = _ble.connectToDevice(id: myPeripheral.Id).listen(null);
await _connectionStateUpdateSubscription?.cancel();
...

void _onConnectDevice(ConnectionStateUpdate state) {
    _logger.i(state.connectionState);
  }
```

**Expected behavior**
The ```connectedDeviceStream```should update twice when i disconnect a peripheral manually. The first update should be with the state ```DeviceConnectionState.disconnecting```, the second with ```DeviceConnectionState.disconnected```

**Smartphone / tablet**
 - Device: Nokia 7.1
 - OS: Android 10
 - Package version: 2.5.1

**Peripheral device**
- Vendor, model: BM832, nRF52832 Bluetooth 5 Module
- Does it run a custom firmware / software: yes

**Additional context**
Shouldn't the  ```connectedDeviceStream``` get also updated after i manually disconnect a device?

### PR DESCRIPTION
**Pull request by remonh87**
Fixes #118

### CODE REVIEW

**Comment by remonh87** [comment]
@werediver  now Android is compliant with the behaviour of the iOS implementation

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.5.3
```
Comment (werediver) [inline CHANGELOG.md:1]: Empty line after a headline?

File: android/src/test/kotlin/com/signify/hue/flutterreactiveble/ble/DeviceConnectorTest.kt [hunk 2]
```diff
@@ -31,12 +32,12 @@ class DeviceConnectorTest {
     private lateinit var device: RxBleDevice
 
     @MockK
-    private lateinit var connectionQueue: com.signify.hue.flutterreactiveble.ble.ConnectionQueue
+    private lateinit var connectionQueue: ConnectionQueue
```
Comment (werediver) [inline android/src/test/kotlin/com/signify/hue/flutterreactiveble/ble/DeviceConnectorTest.kt:35]: 👍

**Review by werediver — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (remonh87) [reply CHANGELOG.md:1]: I feel this is a recurring theme ... 🤔 . Fixed it :)

**Pull request merged**